### DISCUSSION TITLE
Add a map to a document

### DISCUSSION THREAD

**Discussion by lucabalsanelli** [mikefarah/yq#1801] [Q&A]
Hi,

  I want to add a map (which maps a string to another string, e.g. `fff: bar`) to the following

```yaml
aaa: hello
bbb: world
ccc: foo
```

so I'm using

```bash
document='aaa: hello
bbb: world
ccc: foo'
yq '. = . + {fff: bar}' <<<"$document"
```
but I get **Error: 1:10: invalid input text "fff: bar}"**.

If I add a map of numbers, it works

```bash
document='aaa: hello
bbb: world
ccc: foo'
yq '. = . + {111: 222}' <<<"$document"
```

outputs

```
aaa: hello
bbb: world
ccc: foo
111: 222
```

What am I missing?

Thanks,
Luca

**Comment by lucabalsanelli** [comment]
Hi,
  the solution is to add quotes around the `fff` key and its value (source [Add yaml's map to a document using yq](https://stackoverflow.com/a/77188099/3279219))
```bash
document='aaa: hello
bbb: world
ccc: foo'
yq '. += {"fff": "bar"}' <<<"$document"
```
outputs
```yaml
aaa: hello
bbb: world
ccc: foo
fff: bar
```

**Reply by mikefarah** [reply]
Note that you can also do:

```
yq '.fff = "bar"' file.yaml
```
